src/components/results-list/ResultsDataFetcher.tsx

```diff
@@ -71,16 +72,29 @@ const ResultsDataFetcher: FC = () => {
     return readableResults;
   };
 
+  /**
+   * Extracts and returns basic patient information from patient data structures and cDNA changes from observation data structures.
+   * @param patients an array of patients
+   * @param observations an array of patient observations
+   */
   const createReadableResults = (patients: Patient[], observations: Observation[]) => {
-    let readableResults: parsedResultsModel = [];
+    let readableResults: ParsedResultsModel = [];
 
     // extract the required data and store in a trimmed down data structure
     patients.forEach((patient) => {
-      if (!patient.id || !patient.name || !patient.name[0].given) {
+      if (
+        !patient.id ||
+        !patient.identifier ||
+        !patient.identifier[0].value ||
+        !patient.name ||
+        !patient.name[0].given ||
+        !patient.name[0].family
+      ) {
```

Comment:
We could make this a bit more clear by `?` and also using !(with && and && linking). so maybe something like this?
```suggestion
      if (!(
        patient.id &&
        patient.identifier?.get(0)?.value &&
        patient.name?.get(0)?.given &&
        patient.name?.get(0)?.family
      )) {
```